src/components/Modal/readme.md

```diff
@@ -708,3 +708,276 @@ class FormModal extends React.Component {
+            </div>
+
+            <Field
+                className="rainbow-m-bottom_medium"
+                style={styles.input}
+                component={Textarea}
+                name="description"
+                label="Description"
+                placeholder="Add note"
+            />
+        </form>
+    );
+}
+
+const Form = reduxForm({
+    form: 'make-reservation-form',
+    validate,
+    touchOnBlur: false,
+})(ReservationForm);
+
+class FormModal extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            isOpen: false,
+            options: null,
+            initialValues: {
+                date: new Date(),
+                time: '00:00',
+                email: '',
+                country: '',
+                description: '',
+            },
+        };
+        this.handleOnClick = this.handleOnClick.bind(this);
+        this.handleOnClose = this.handleOnClose.bind(this);
+        this.handleSubmit = this.handleSubmit.bind(this);
+    }
+
+    handleOnClick() {
+        return this.setState({
+            isOpen: true,
+        });
+    }
+
+    handleOnClose() {
+        return this.setState({
+            isOpen: false,
+            options: null,
+        });
+    }
+
+    handleSubmit(values) {
+        console.log(values);
+        this.setState({
+            options: null,
+        });
+    }
+
+    render() {
+        const { isOpen, initialValues, isLoading, options } = this.state;
+        return (
+            <div>
+                <Button
+                    id="button-12"
+                    variant="neutral"
+                    label="Book Now!!"
+                    onClick={this.handleOnClick}
+                    variant="brand"
+                />
+                <Modal
+                    id="modal-12"
+                    title="Add your reservation"
+                    isOpen={isOpen}
+                    onRequestClose={this.handleOnClose}
+                    footer={
+                        <div className="rainbow-flex rainbow-justify_end">
+                            <Button
+                                form="redux-form-id"
+                                className="rainbow-m-right_large"
+                                label="Cancel"
+                                variant="neutral"
+                                onClick={this.handleOnClose}
+                            />
+                            <Button
+                                form="redux-form-id"
+                                label="Save"
+                                variant="brand"
+                                type="submit"
+                            />
+                        </div>
+                    }
+                >
+                    <Form
+                        onSubmit={this.handleSubmit}
+                        onSearch={this.search}
+                        isLoading={isLoading}
+                        options={options}
```

Comment:
Form component does not receive options, isLoading, and onSearch props, then clean up all this props